### PR TITLE
Add new OAuthApplicationMgtListener to handle OAuth application management events

### PR DESCRIPTION
**Pull request by thanujalk**
### Proposed changes in this pull request

Add new OAuthApplicationMgtListener to handle OAuth application management events.
- Part of https://github.com/wso2/product-is/issues/10335

### When should this PR be merged

- Immediate 

### Follow up actions

- N/A

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthApplicationMgtListener.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    void doPreUpdateConsumerApplication(OAuthConsumerAppDTO consumerAppDTO) throws IdentityOAuthAdminException;
+
+    /**
+     * Pre-listener for the update consumer app state.
+     *
+     * @param consumerKey consumer key.
+     * @param newState    new state.
+     * @throws IdentityOAuthAdminException in case of failure.
+     */
+    void doPreUpdateConsumerAppState(String consumerKey, String newState) throws IdentityOAuthAdminException;
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthApplicationMgtListener.java:58]:
```suggestion
    void doPreUpdateConsumerApplicationState(String consumerKey, String newState) throws IdentityOAuthAdminException;
```
for consistency

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthApplicationMgtListener.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+package org.wso2.carbon.identity.oauth.listener;
+
+import org.wso2.carbon.identity.oauth.IdentityOAuthAdminException;
+import org.wso2.carbon.identity.oauth.dto.OAuthConsumerAppDTO;
+
+/**
+ * Listener interface for OAuth application management CRUD operations.
+ */
+public interface OAuthApplicationMgtListener {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthApplicationMgtListener.java:27]: Wouldn't it be worth to have post events too?

**Review by IsuraD — APPROVED** [review]

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthApplicationMgtListener.java (hunk 2 above)
Comment (thanujalk) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthApplicationMgtListener.java:27]: Yes. Will be added later.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthApplicationMgtListener.java (hunk 1 above)
Comment (thanujalk) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/OAuthApplicationMgtListener.java]: updated

**Pull request merged**